ddtrace/appsec/processor.py

```diff
@@ -188,8 +204,12 @@ def on_span_finish(self, span):
                 data[_Addresses.SERVER_RESPONSE_HEADERS_NO_COOKIES] = _no_cookies(response_headers)
 
         log.debug("[DDAS-001-00] Executing AppSec In-App WAF with parameters: %s", data)
-        res = self._ddwaf.run(data)  # res is a serialized json
+        res = self._ddwaf.run(data, self._waf_timeout)  # res is a serialized json
         if res is not None:
+            allowed = self._rate_limiter.is_allowed(span.start_ns)
```

Comment:
wouldn't we want this to run first/right after the check for web type?

is this meant to rate limit how often we run the waf on requests?

Reply:
The goal here is to limit the number of collected attacks but not limit WAF executions. Limiting WAF executions would give a slightly different result - I will add some comments to clarify.